sklearn/datasets/base.py

```diff
@@ -823,3 +830,48 @@ def _pkl_filepath(*args, **kwargs):
         basename += py3_suffix
     new_args = args[:-1] + (basename + ext,)
     return join(*new_args)
+
+
+def _sha256(path):
+    """Calculate the sha256 hash of the file at path."""
+    sha256hash = hashlib.sha256()
+    chunk_size = 8192
+    with open(path, "rb") as f:
+        while True:
+            buffer = f.read(chunk_size)
+            if not buffer:
+                break
+            sha256hash.update(buffer)
+    return sha256hash.hexdigest()
+
+
+def _fetch_url(url, path, checksum):
+    """Fetch a dataset and check the SHA256 checksum
+
+    Fetch a dataset pointed by url, save into path and ensure its integrity
+    based on the SHA256 Checksum of the downloaded file.
+
+    Parameters
+    -----------
+    URL : string
+        URL to fetch the download from.
+
+    path : string
+        Path to save the file to.
+
+    checksum : string
+        SHA256 checksum to verify against the data
+
+    """
+
+    download(url, path)
+    if checksum != _sha256(path):
+        raise IOError("{} has an SHA256 hash differing from expected, "
+                      "file may be corrupted.".format(path))
+
+
+def _fetch_remote(remote, path=None):
```

Comment:
why are we not checking the checksum here? And from the docstring it's not clear what the difference to _fetch_url is...